compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -4546,14 +4546,11 @@ private void validateReferencedFunction(Location pos, BAttachedFunction func, Sy
             return;
         }
 
-        // Service typing does not consider resource methods when type checking.
-        if (Symbols.isResource(func.symbol)) {
-            return;
-        }
-
         // There must be an implementation at the outer level, if the function is an interface.
```

Comment:
Not related to your change, but this comment isn't quite correct. Let's remove/update separately.